### ISSUE TITLE
Error: Couldn't get access token

### ISSUE THREAD

**Issue by hiifeng** [m1guelpf/chatgpt-telegram#71]
Using the latest v0.2.4 version, the bot replies:
Error: Couldn't get access token: failed to decode response: invalid character '<' looking for beginning of value
Does anyone know what is the reason? Is it because OpenAI added cloudflare?

**Comment by R0uter** [comment]
same here with the docker img

**Comment by alastone** [comment]
同样的问题

**Comment by a1gx** [comment]
same question

**Comment by bwmgd** [comment]
same question

**Comment by jchby** [comment]
same error

**Comment by tianlichunhong** [comment]
I think it is blocked by OpenAI. It doesn't want to be used too much.

**Comment by Wandering-Consciousness** [comment]
Same issue here. Something changed on the OpenAI side in the last day or so.

**Comment by MichaelCutro** [comment]
@dhilman Any updates on this? Going to try and run it on a vm tomorrow but I am getting the same error on M1

**Comment by MichaelCutro** [comment]
The error is coming from line 148 of chatgpt.go -  "failed to decode response: invalid character '<' looking for beginning of value" suggests that the response received from the OpenAI API when the bot attempts to retrieve an access token is not a valid JSON object, but instead contains an HTML page or other invalid content. This could be caused by a number of issues, such as a misconfigured server or endpoint, a network connectivity problem, or an issue with the bot's code or configuration.

**Comment by gaeduf** [comment]
so how can we solve this problem?

**Comment by alastone** [comment]
> 那么我们如何解决这个问题呢？

It's caused by the addition of cloudflare protection

**Comment by llegomark** [comment]
Just a few hours ago, OpenAI integrated Cloudflare security into their system.

**Comment by BlueSkyLT** [comment]
Same issue.

**Comment by msilenkov** [comment]
Может быть тогда стоит подключиться вот сюда? 
https://beta.openai.com/playground 

:)

**Comment by w276211640** [comment]
![image](https://user-images.githubusercontent.com/85927003/207023175-24b82260-f3c5-4e29-8a18-58c5c92a9a9e.png)
you can this  
![image](https://user-images.githubusercontent.com/85927003/207023269-f716c3f1-282f-4496-8cfb-c55be686c904.png)

**Comment by Xiefengshang** [comment]
It seems to be the problem of `Cloudflare Captcha`.  How about adding browser plugin?

**Comment by llegomark** [comment]
Aside from Cloudflare, ChatGPT no longer outputs long-form content, if you hit the limit, it will throw a network error, or the CLI Response not in correct format! peer closed connection without sending complete message body (incomplete chunked read)

**Comment by hiifeng** [comment]
> 



> Aside from Cloudflare, ChatGPT no longer outputs long-form content, if you hit the limit, it will throw a network error, or the CLI Response not in correct format! peer closed connection without sending complete message body (incomplete chunked read)

Yes, you are quite right. I also found this problem, when I use https://chat.openai.com/chat, "network error" also occurs. I am not familiar with the go language, is there a solution?

**Comment by gitk01n** [comment]
> you are quite right ，inside cf_cf_clearance token

**Comment by taxueqinyin** [comment]
get refreshAccessToken is 403 forbidden,because open ai does man-machine validation on the request,i thought we should find a way to around it

**Comment by waltcow** [comment]
https://github.com/acheong08/ChatGPT/issues/261

**Comment by m1guelpf** [comment]
Closing in favor of #69 to keep everything in one place

**Comment by MichaelCutro** [comment]
> > 
> 
> > Aside from Cloudflare, ChatGPT no longer outputs long-form content, if you hit the limit, it will throw a network error, or the CLI Response not in correct format! peer closed connection without sending complete message body (incomplete chunked read)
> 
> Yes, you are quite right. I also found this problem, when I use https://chat.openai.com/chat, "network error" also occurs. I am not familiar with the go language, is there a solution?
> 
> > **[Just a moment...](https://chat.openai.com/chat)**

I'm not very familiar with Go either but I have built bots in the past and gotten past Cloudflare with [Puppeteer-extra ](https://github.com/berstend/puppeteer-extra/tree/master/packages/puppeteer-extra-plugin-stealth) I don't think this is a Cloudflare problem though rather the incomplete chunked read that @llegomark mentioned above
